add blog post about release 0.1.54 (#17)
ComplianceAsCode/ComplianceAsCode.github.io: _posts/2021-02-05-release-0.1.54.md

---
layout: post

> title: "Releasing Compliance as code / Content 0.1.54"

Reviewer: ComplianceAsCode

categories: template
author: Vojtěch Polášek
---

I am happy to announce that Compliance as code / Content v0.1.54 [has been released](https://github.com/ComplianceAsCode/content/releases/tag/v0.1.54) on February 3rd. This is the first version released in 2021 and it brings many interesting improvements for users as well as content developers.

It has been also quite a long time since a release has been accompanied by a blog post. We hope that we can change it for future releases. As you can read below, there are many things which deserve simply more than one line in release notes. So what has changed since 0.1.53?

## Highlight of this release

This release introduces centralised policy definitions. This feature allows to use an alternative approach to profile definition. Until now, a profile has been defined as a linear list of rules. And if you wanted to have a similar profile (policy interpretation) for multiple products, you would have to keep separate profile files for each product.

> Centralised policy definitions try to solve these problems by allowing higher abstraction when defining a profile. They allow to define a policy with multiple requirements, where each requirement can translate into multiple rules. We hope that this will improve user experience of profile maintenance and reduce redundant content in the project.

Reviewer: ```suggestion
Centralised policy definitions try to solve these problems by allowing higher abstraction when defining a profile. They allow to define a policy with multiple requirements, where each requirement can translate into multiple rules. We hope that this will streamline profile maintenance and reduce redundant content in the project.
```

This feature definitely deserves more space than two paragraphs, so expect a special blog post about it. Or if you are super curious, you can read description of the [pull request](https://github.com/ComplianceAsCode/content/pull/6499).

## What's new in the content?

> One more word about policy definitions; it is not just a proof of concept, there already exists a set of profiles which is defined through this mean. The new release brings ANSSI BP-028 profiles for Red Had Enterprise Linux 7 and REd Hat Enterprise Linux 8. There are profiles for Minimal, Intermediary, Enhanced and High hardening levels.

Reviewer: ```suggestion
One more word about policy definitions; it is not just a proof of concept, there already exists a set of profiles which is defined through this mean. The new release brings ANSSI BP-028 profiles for Red Hat Enterprise Linux 7 and Red Hat Enterprise Linux 8. There are profiles for Minimal, Intermediary, Enhanced and High hardening levels.
```

> One more word about policy definitions; it is not just a proof of concept, there already exists a set of profiles which is defined through this mean. The new release brings ANSSI BP-028 profiles for Red Had Enterprise Linux 7 and REd Hat Enterprise Linux 8. There are profiles for Minimal, Intermediary, Enhanced and High hardening levels.

Reviewer 2: Capitalization in "Red Hat Enterprise Linux 8"

Second thing is - we have some confidence in the Minimal Intermediary and Enhanced. But the "High" profile is still work in progress. This can be expressed here as well.

> While  creating the profiles, we are communicating with representatives of ANSSI organization, so that the result is as precise as possible. You can read more about ANSSI BP-028 [here](https://www.ssi.gouv.fr/en/guide/configuration-recommendations-of-a-gnulinux-system/).

Reviewer 2: Let's drop this paragraph :) We are in touch, but we are not in the state to say "definitive ANSSI profile", yet :) Let's leave it for later, possibly for specialized blog post.

A new revision v3r2 of DISA STIG for Red Hat Enterprise Linux 7 [has been released](https://public.cyber.mil/announcement/stig-update-january-2021-quarterly-release/). We went carefully over new changes and reflected them in the appropriate profile.

We were not only adding, but also removing and cleaning. This release no longer contains content for Red Hat Enterprise Linux 6 which entered its Extended Life Phase on November 30, 2020.

The project documentation received a major upgrade. It has been successfully transformed from set of markdown files into interactive sites hosted at [readthedocs.org](https://readthedocs.org/). This makes it much easier to read and update it. You can currently look at:

- Developer guide

- Release tools documentation

- SSG Test Suite developer guide

- Jinja2 macro reference

- Python module reference

> 

Reviewer: I think this fits better into "What's new for content developers?" than here because the docs contain mostly developer guide and developer docs.

You can read more about this effort in the [special blog post](https://complianceascode.github.io/template/2021/01/19/developer-documentation.html) or head directly to the [documentation](https://complianceascode.readthedocs.io/en/latest/).

> In OCP, most of the work we've been doing has been related to the unreleased OpenShift CIS benchmark. This benchmark is closely based from the Kubernetes CIS benchmark, but includes the knobs and handles that OpenShift exposes to configure and thus secure your cluster. There are two profiles available, but they're meant to be ran together. cis & cis-node. The profile called cis checks the Kubernetes API objects directly, and verifies that they contain the expected values. The cis-node profile is meant to run on the node itself and will do more low-level checks, such as file permissions, or contents on specific files on the node. They, together, encompass the whole benchmark.

Reviewer: ```suggestion
In OCP, most of the work we've been doing has been related to the unreleased OpenShift CIS benchmark. This benchmark is closely based from the Kubernetes CIS benchmark, but includes the knobs and handles that OpenShift exposes to configure and thus secure your cluster. There are two profiles available, but they're meant to be ran together.  The profile called cis checks the Kubernetes API objects directly, and verifies that they contain the expected values. The cis-node profile is meant to run on the node itself and will do more low-level checks, such as file permissions, or contents on specific files on the node. They, together, encompass the whole benchmark.
```

## What's new for content developers?

> The rule template for checking of file permissions received an interesting update. Until now, it was possible to specify only the exact set of permissions to check for. If the file permissions were different, even stricter (and therefore more secure), the rule would be marked as "failed". Now it is possible to say that more restrictive permissions are OK and the rule should "pass". You can check the exact parameters [here](https://complianceascode.readthedocs.io/en/latest/manual/developer/06_contributing_with_content.html#templating).

Reviewer: ```suggestion
The rule template for checking file permissions received an interesting update. Until now, it was possible to specify only the exact set of permissions to check for. If the file permissions were different, even stricter (and therefore more secure), the rule would be marked as "failed". Now it is possible to say that more restrictive permissions are OK and the rule should "pass". You can check the exact parameters [here](https://complianceascode.readthedocs.io/en/latest/manual/developer/06_contributing_with_content.html#file-permissions).
```

> Our test suite received a nice update as well. You can now test your rules easier and faster thanks to two wrapper scripts which simplify usage of Podman backend. To get more information about Podman and test suite, head [here](https://complianceascode.readthedocs.io/en/latest/). If you are interested about details of this change, you can read the [PR description](https://github.com/ComplianceAsCode/content/pull/6405).

Reviewer: Show us an example.

> The declaration syntax of CPE Items changed and each product's CPE Dictionaries is built only with the necessary items. A lot of boilerplate around CPE Items was removed and simplified to a yaml syntax.

Reviewer: ```suggestion
The declaration syntax of CPE Items changed and each product's CPE Dictionaries is built only with the necessary items. A lot of boilerplate code around CPE Items was removed and simplified to a yaml syntax.
```

> The templating system has been simplified. Each template now has its separate folder. Adding a new template now requires just placing right files into right folders and you are ready to go.

Reviewer: As a reader I have no idea what are right files and where are right folders. Please either make it more high level or more specific.

Reviewer: Maybe show us an example
